library/src/main/java/com/novoda/viewpageradapter/ViewPagerAdapterState.java

```diff
@@ -40,8 +40,11 @@ private static ViewPagerAdapterState from(Parcel in) {
     }
 
     private static SparseIntArray extractIdsFrom(Bundle bundle) {
-        SparseIntArray output = new SparseIntArray();
         int[] ids = bundle.getIntArray(KEY_VIEW_IDS);
+        if (ids == null) {
```

Comment:
There were no previous null checks in here, so if `ids` was null, there would have been a `NullPointerException` when trying to get the length of the array.